gr-uhd/lib/usrp_block_impl.cc

```diff
@@ -741,3 +741,9 @@ void usrp_block_impl::_cmd_handler_dspfreq(const pmt::pmt_t& dspfreq,
 
     _update_curr_tune_req(new_tune_request, chan);
 }
+
+::uhd::dict<std::string, std::string> usrp_block_impl::get_usrp_info(size_t chan)
+{
+    chan = _stream_args.channels[chan];
+    return _dev->get_usrp_rx_info(chan);
```

Comment:
The `multi_usrp` class has both `get_usrp_tx_info` and `get_usrp_rx_info` functions, with some different information between each. The separate implementations in `usrp_source` and `usrp_sink` were intentional for this reason. Consolidating these into `usrp_block` removes getting TX-specific info.

Reply:
That's a very good reason to keep them separate. But also, it's a very good reason to update these function names.
I suggest to add `get_usrp_rx_info` to `usrp_source` and `get_usrp_tx_info` to `usrp_sink`. Further, I suggest to keep `get_usrp_info` in the base class, deprecate it and log a warning everytime it is called. Alternatively, we can remove it. But I doubt this approach receives approval without deprecation for at least one release cycle.
Anyways, I'd consider it a bug that the function does not fully describe what it does and needs to be fixed.

Reply:
The `get_usrp_*x_info` solution, with deprecation of the old function name, sounds like a good plan.

Reply:
Every other function in these blocks has the same name for the sink and source (like set gain, set freq) and is declared in usrp_block, and then defined in the sink or source. I don't see a necessity to change that here.